sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -304,33 +308,78 @@ export class ManagementClient extends LinkEntity {
-        abortSignal: options.abortSignal,
-        requestName: options.requestName,
-        timeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions) / 1000,
-        delayInSeconds:
-          options.retryOptions &&
-          options.retryOptions.retryInterval &&
-          options.retryOptions.retryInterval >= 0
-            ? options.retryOptions.retryInterval / 1000
-            : undefined
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+
+          let waitTimer: any;
+          const actionAfterTimeout = () => {
+            const address = this._mgmtReqResLink!.receiver.address || "address";
+            const desc: string =
+              `The request with message_id "${request.message_id}" to "${address}" ` +
+              `endpoint timed out. Please try again later.`;
+            const e: Error = {
+              name: "OperationTimeoutError",
+              message: desc
+            };
+            return reject(translate(e));
+          };
+
+          waitTimer = setTimeout(
+            actionAfterTimeout,
+            getRetryAttemptTimeoutInMs(options!.retryOptions)
+          );
+
+          const initOperationStartTime = Date.now();
+          log.mgmt(
+            "[%s] Acquiring lock to get the management req res link.",
+            this._context.connectionId
+          );
+          await defaultLock.acquire(this.managementLock, () => {
+            return this._init().catch((err: Error) => {
+              clearTimeout(waitTimer);
+              reject(translate(err));
+            });
+          });
+          const initOperationEndTime = Date.now();
+
+          if (!options) {
+            options = {};
+          }
+
+          const operationTimeoutInMs = initOperationEndTime - initOperationStartTime;
+          const sendRequestOptions: SendRequestOptions = {
+            abortSignal: options.abortSignal,
+            requestName: options.requestName,
+            timeoutInSeconds: operationTimeoutInMs / 1000
+          };
+          resolve((await this._mgmtReqResLink!.sendRequest(request, sendRequestOptions)));
+        });
+
+      const jitterInSeconds = randomNumberFromInterval(1, 4);
+      const maxRetries = options.retryOptions && options.retryOptions.maxRetries;
+      const delayInSeconds =
+        options.retryOptions &&
+        options.retryOptions.retryInterval &&
+        options.retryOptions.retryInterval >= 0
+          ? options.retryOptions.retryInterval / 1000
+          : Constants.defaultDelayBetweenOperationRetriesInSeconds;
+      const config: RetryConfig<void> = {
+        operation: sendOperationPromise,
+        connectionId: this._context.connectionId,
+        operationType:
+          request.to && request.to === Constants.cbsEndpoint
+            ? RetryOperationType.cbsAuth
+            : RetryOperationType.management,
+        maxRetries: maxRetries,
```

Comment:
I think you'll need to set this to `maxRetries + 1` to account for the 1st attempt that should always be made. Otherwise if the customer sets `maxRetries` to 0, you won't send the request.

Reply:
@chradek I believe the `retry()` function does the +1 inside

Reply:
Oh, I missed when that change happened. I’ll have to update that in the receiver refactor as well then.